File: jogl-samples/src/framework/Semantic.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package framework;

/**
 *
 * @author gbarbieri
 */
public class Semantic {

    public static class Attr {

        public static final int POSITION = 0;
        public static final int NORMAL = 1;
        public static final int COLOR = 3;
        public static final int TEXCOORD = 4;
        public static final int DRAWID = 5;
    }

    public static class Frag {

        public static final int COLOR = 0;
        public static final int RED = 0;
        public static final int GREEN = 1;
        public static final int BLUE = 2;
        public static final int ALPHA = 0;
    }

    public static class Uniform {

        public static final int MATERIAL = 0;
        public static final int TRANSFORM0 = 1;
        public static final int TRANSFORM1 = 2;
        public static final int INDIRECTION = 3;
        public static final int CONSTANT = 0;
        public static final int PERFRAME = 1;
        public static final int PERPASS = 2;
        public static final int LIGHT = 3;
    }

    public static class Object {

        public static final int VAO = 0;
        public static final int VBO = 1;
        public static final int IBO = 2;
        public static final int SIZE = 3;
    }
}

File: src/main/kotlin/oglSamples/framework/semantic.kt
package oglSamples.framework

/**
 * Created by GBarbieri on 27.03.2017.
 */

object semantic {

    object buffer {
        val STATIC = 0
        val DYNAMIC = 1
    }

    object uniform {
        val MATERIAL = 0
        val TRANSFORM0 = 1
        val TRANSFORM1 = 2
        val INDIRECTION = 3
        val CONSTANT = 0
        val PER_FRAME = 1
        val PER_PASS = 2
        val LIGHT = 3
    }

    object sampler {
        val DIFFUSE = 0
        val POSITION = 4
        val TEXCOORD = 5
        val COLOR = 6
    }

    object image {
        val DIFFUSE = 0
        val PICKING = 1
    }

    object attr {
        val POSITION = 0
        val NORMAL = 1
        val COLOR = 3
        val TEXCOORD = 4
        val DRAW_ID = 5
    }

    object vert {
        val POSITION = 0
        val COLOR = 3
        val TEXCOORD = 4
        val INSTANCE = 7
    }

    object frag {
        val COLOR = 0
        val RED = 0
        val GREEN = 1
        val BLUE = 2
        val ALPHA = 0
    }

    object renderbuffer {
        val DEPTH = 0
        val COLOR0 = 1
    }

    object storage {
        val VERTEX = 0
    }
}
